packages/queued-request-controller/tests/QueuedRequestMiddleware.test.ts

```diff
@@ -0,0 +1,425 @@
+  jsonrpc: '2.0' as const,
+  origin: 'example.com',
+  networkClientId: 'mainnet',
+};
+
+describe('createQueuedRequestMiddleware', () => {
+  it('throws if not provided an origin', async () => {
+    const messenger = buildMessenger();
+    const middleware = createQueuedRequestMiddleware({
+      messenger,
+      useRequestQueue: () => false,
+    });
+    const req = {
+      id: '123',
+      jsonrpc: '2.0',
+      method: 'anything',
+      networkClientId: 'anything',
+    };
+
+    await expect(
+      () =>
+        new Promise((resolve, reject) =>
+          middleware(req as any, {} as any, resolve, reject),
+        ),
+    ).rejects.toThrow("Request object is lacking an 'origin'");
+  });
+
+  it('throws if not provided an networkClientId', async () => {
+    const messenger = buildMessenger();
+    const middleware = createQueuedRequestMiddleware({
+      messenger,
+      useRequestQueue: () => false,
+    });
+    const req = {
+      id: '123',
+      jsonrpc: '2.0',
+      method: 'anything',
+      origin: 'anything',
+    };
+
+    await expect(
+      () =>
+        new Promise((resolve, reject) =>
+          middleware(req as any, {} as any, resolve, reject),
+        ),
+    ).rejects.toThrow("Request object is lacking a 'networkClientId'");
+  });
+
+  it('should not enqueue the request when useRequestQueue is false', async () => {
+    const messenger = buildMessenger();
+    const middleware = createQueuedRequestMiddleware({
+      messenger,
+      useRequestQueue: () => false,
+    });
+    const mocks = buildMocks(messenger);
+
+    await new Promise((resolve, reject) =>
+      middleware({ ...requestDefaults }, {} as any, resolve, reject),
+    );
+
+    expect(mocks.enqueueRequest).not.toHaveBeenCalled();
+  });
+
+  it('should not enqueue the request when there is no confirmation', async () => {
+    const messenger = buildMessenger();
+    const middleware = createQueuedRequestMiddleware({
+      messenger,
+      useRequestQueue: () => true,
+    });
+    const mocks = buildMocks(messenger);
+
+    const req = {
+      ...requestDefaults,
+      method: 'eth_chainId',
+    };
+
+    await new Promise((resolve, reject) =>
+      middleware(req, {} as any, resolve, reject),
+    );
+
+    expect(mocks.enqueueRequest).not.toHaveBeenCalled();
+  });
+
+  describe('confirmations', () => {
+    it('should resolve requests that require confirmations for infura networks', async () => {
+      const messenger = buildMessenger();
+      const middleware = createQueuedRequestMiddleware({
+        messenger,
+        useRequestQueue: () => true,
+      });
+      const mocks = buildMocks(messenger);
+
+      const req = {
+        ...requestDefaults,
+        method: 'eth_sendTransaction',
+      };
+
+      await new Promise((resolve, reject) =>
+        middleware(req, {} as any, resolve, reject),
+      );
+
+      expect(mocks.enqueueRequest).toHaveBeenCalled();
+      // infura networks do not use getNetworkClientById
```

Comment:
could we assert here that `NetworkController:setProviderType` has been called?

Reply:
negative - switch isnt called in this case. More importantly, this test isnt asserting any behaviour relating to switching chain - it just serves as the base case, ensuring that requests with confirmations will resolve without issue. This block: ` describe('requiring switch', () => {` is where the tests handling switching are located.